spec/mongoid/matchable/nor_spec.rb

```diff
@@ -0,0 +1,199 @@
+        let(:matches) do
+          matcher._matches?(
+            [ { af: "Sir" }, { af: "King" } ]
+          )
+        end
+
+        let(:target) do
+          ArrayField.new(af: ['Boo'])
+        end
+
+        it "returns true" do
+          expect(matches).to be true
+        end
+      end
+
+      context "when there are no criteria" do
+
+        it "returns false" do
+          expect(matcher._matches?([])).to be false
+        end
+      end
+
+      # $nor with $not is a double negation.
+      # Whatever the argument of $not is is what the overall condition
+      # is looking for.
+      context "when the expression is a $not" do
+
+        let(:matches) do
+          matcher._matches?([ { title: {:$not => /Foobar/ } }])
+        end
+
+        context "when the value does not match $not argument" do
+
+          let(:target) do
+            Person.new(title: 'test')
+          end
+
+          it "returns false" do
+            expect(matches).to be false
+          end
+        end
+
+        context "when the value matches $not argument" do
+
+          let(:target) do
+            Person.new(title: 'Foobar baz')
+          end
+
+          it "returns true" do
+            expect(matches).to be true
+          end
+        end
+      end
+    end
+
+    context "when provided a complex expression" do
+
+      context "when none of the model values match criteria values" do
+
+        let(:matches) do
+          matcher._matches?(
+            [
+              { title: { "$in" => [ "Sir", "Madam" ] } },
+              { title: "King" }
+            ]
+          )
+        end
+
+        let(:target) do
+          Person.new(title: 'Queen')
+        end
+
+        it "returns true" do
+          expect(matches).to be true
+        end
+      end
+
+      context "when there is a matching value" do
+
+        let(:matches) do
+          matcher._matches?(
+            [
+              { title: { "$in" => [ "Prince", "Madam" ] } },
+              { title: "King" }
+            ]
+          )
+        end
+
+        let(:target) do
+          Person.new(title: 'Prince')
+        end
+
+        it "returns false" do
+          expect(matches).to be false
+        end
+      end
+
+      context "when expression contain multiple fields" do
+
+        let(:matches) do
+          matcher._matches?(
+            [
+              { title: "Sir", age: 23 },
+              { title: "King", age: 100 }
+            ]
+          )
+        end
+
+        context 'and model has different values in all of the fields' do
+          let(:target) do
+            Person.new(title: 'Queen', age: 10)
+          end
+
+          it "returns true" do
+            expect(matches).to be true
+          end
+        end
+
+        context 'and model has identical value in one of the fields' do
+          let(:target) do
+            Person.new(title: 'Queen', age: 23)
+          end
+
+          it "returns false" do
```

Comment:
I don't think this is correct (and by extension, I think the implementation of `_matches?`) is incorrect. The method should return true if none of the queries in the array are true, which is the case since neither `{ title: 'King', age: 100 }` or `{ title: 'Sir', age: 23 }` matches `{ title: 'Queen', age: 100 }`. You can try this out on the shell with the following code and see that the document is actually returned:

```javascript
db.foo.insert({ title: 'Queen', age: 23 })
db.foo.find({ '$nor': [
    { title: 'Sir', age: 23 },
    { title: 'King', age: 100 },
]})
```

See my comment above for how I think `_matches?` should be implemented to fix this.

Reply:
Right, good catch.

Reply:
Good find. @p-mongo are you taking on @saghm's comments or should I slot time to make changes?

Reply:
I fixed it.